docs/book/content/theory/firms.md

```diff
@@ -10,14 +10,24 @@ The production side of the `OG-Core` model is populated by a unit measure of ide
 
   ```{math}
   :label: EqFirmsCESprodfun
-    Y_t = F(K_t, K_{g,t}, L_t) \equiv Z_t\biggl[(\gamma)^\frac{1}{\varepsilon}(K_t)^\frac{\varepsilon-1}{\varepsilon} + (\gamma_{g})^\frac{1}{\varepsilon}(K_{g,t})^\frac{\varepsilon-1}{\varepsilon} + (1-\gamma-\gamma_{g})^\frac{1}{\varepsilon}(e^{g_y t}L_t)^\frac{\varepsilon-1}{\varepsilon}\biggr]^\frac{\varepsilon}{\varepsilon-1} \quad\forall t
+    \begin{split}
```

Comment:
Do we need these cases?  And if so, is if correct call one the `K_g=0` case?  If K_g = 0, the first function is defined.  It seems that the second case is really one that ignores K_g.  But then the first case captures that with `gamma_g=0`.

Reply:
@jdebacker. It is a little different than your restatement above. I saw two options to go with this. I'll label the options here (A) allow `K_g >= 0` or (B) force `K_g > 0` always. I went with option (A), but (B) has some advantages. I like `K_g` to have the option of being zero. We just have to know there are some caveats if `K_g=0` and `varepsilon=1` (Cobb-Douglas production).

**A) Allow `K_g > 0` and `K_g = 0`.**
A1) (good) When `K_g > 0`, everything is straightforward with the CES production function for any value of `varepsilon >= 1`.
A2) (good) When `K_g = 0` and `varepsilon > 1` (not Cobb-Douglas) everything is straightforward and consistent because the terms in the parentheses of the CES production function are additive. The `K_g = 0` does not zero out production.
A3) (a little weird) When `K_g=0` and `varepsilon = 1`, the `K_g=0` zeros out output and breaks the model. The only way I could come up with solving this case is having a different Cobb-Douglas production function that excludes `K_g`. In this case where `varepsilon=1` it would be a little weird to do an experiment like the effect of `K_g` being positive then going to zero or vice versa because the production function would fundamentally change in those cases. But I think that one caveat is worth allowing `K_g` to be zero.
A4) I would operationalize the switching rule as the following. If `varepsilon = 1` and `K_g < some_epsilon`, then switch to the Cobb-Douglas production function with only `K` and `L`.

**B) Force `K_g > 0`**
This approach gets rid of the problem associated with case (A3). `K_g` could be forced to be greater than some small value. The only drawback is you cannot do any experiments with `K_g=0`.

Let me know what you think.

Reply:
@jdebacker said of case (A3):
> This is only true if `gamma_g > 0` and makes sense for that case. If `gamma_g=0`, then `K_g=0` is fine since it's not an input to production.

So are you suggesting that when `K_g=0` and `varepsilon=1`, we set `gamma_g=0`? That is the same as the different production function I am suggesting.

Reply:
> So are you suggesting that when K_g=0 and varepsilon=1, we set gamma_g=0?

Yes, if we think K_g isn't critical to producing output.


> That is the same as the different production function I am suggesting.

Right, but then all cases nested.  e.g., the model already allows for a labor only production function with gamma=0.  We don't specify a separate case for that and I'm just suggesting that a production function that doesn't rely on K_g be treated the same.

Reply:
@jdebacker. Yes. It is two ways of doing the same thing:
1. When `K_g=0` and `epsilon=1`, set `gamma_g=0`.
2. When `K_g=0` and `epsilon=1`, change the production function to `Y=F(K,L)`.

In both cases, we cannot do a clean experiment of the effect of `baseline: K_g=0` versus `reform: K_g>0` or vice versa. But I am OK with that. Let's go with whatever way you prefer.

Another way of describing the modeling "party foul" of doing either method above is that `gamma_g` is a deep parameter that is supposed to be policy invariant. Ideally we would calibrate it to be something, then let `K_g` move around like a policy parameter. In either of the approaches above, we break that rule out of necessity in the one special case: `K_g=0` and `epsilon=1`.

Reply:
@jdebacker . Another implication of what I am saying above is that, in general, `gamma_g` should be greater than 0, when `K_g=0`. Only when `epsilon=1` and `K_g=0` should `gamma_g=0` or `Y=F(K,L)`.